base/applications/cmdutils/where/strlist.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * PROJECT:     ReactOS WHERE command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Providing string list
+ * COPYRIGHT:   Copyright 2021 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#pragma once
+
+#define str_clone _wcsdup
+
+typedef struct strlist_t
+{
+    LPWSTR *ppsz;
+    unsigned int count;
+} strlist_t;
+#define strlist_default (strlist_t){ NULL, 0 }
+
+static inline void strlist_init(strlist_t *plist)
+{
+    *plist = strlist_default;
+}
+
+static inline LPWSTR strlist_get_at(strlist_t *plist, unsigned int i)
+{
+    return plist->ppsz[i];
+}
+
+static int strlist_add(strlist_t *plist, LPCWSTR psz)
+{
+    LPWSTR *ppsz, clone = str_clone(psz);
+    if (!clone)
+        return 0;
+    ppsz = (LPWSTR *)realloc(plist->ppsz, (plist->count + 1) * sizeof(LPWSTR));
+    if (!ppsz)
+        return 0;
```

Comment:
In case this allocation fails, there is leakage of the allocated "clone" pointer.

Reply:
OK.